pkg/operation/botanist/component/backupentry/backupentry.go

```diff
@@ -159,36 +149,25 @@ func (b *backupEntry) WaitMigrate(ctx context.Context) error {
 // Restore uses the garden client to update the BackupEntry and set the name of the new seed to which it shall be scheduled.
 // If the BackupEntry was deleted it will be recreated.
 func (b *backupEntry) Restore(ctx context.Context, _ *gardencorev1alpha1.ShootState) error {
-	var (
-		backupEntry = &gardencorev1beta1.BackupEntry{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      b.values.Name,
-				Namespace: b.values.Namespace,
-			},
-		}
-		bucketName = b.values.BucketName
-	)
+	bucketName := b.values.BucketName
 
-	if err := b.client.Get(ctx, kutil.Key(b.values.Namespace, b.values.Name), backupEntry); err == nil {
-		bucketName = backupEntry.Spec.BucketName
+	if err := b.client.Get(ctx, kutil.Key(b.values.Namespace, b.values.Name), b.backupEntry); err == nil {
+		bucketName = b.backupEntry.Spec.BucketName
 	} else if client.IgnoreNotFound(err) != nil {
 		return err
 	}
-	return b.createOrUpdate(ctx, backupEntry, b.values.SeedName, bucketName, v1beta1constants.GardenerOperationRestore)
+	return b.reconcile(ctx, b.backupEntry, b.values.SeedName, bucketName, v1beta1constants.GardenerOperationRestore)
 }
 
-func (b *backupEntry) createOrUpdate(ctx context.Context, backupEntry *gardencorev1beta1.BackupEntry, seedName *string, bucketName string, operation string) error {
-	_, err := controllerutil.CreateOrUpdate(ctx, b.client, backupEntry, func() error {
+func (b *backupEntry) reconcile(ctx context.Context, backupEntry *gardencorev1beta1.BackupEntry, seedName *string, bucketName string, operation string) error {
+	_, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, b.client, backupEntry, func() error {
 		metav1.SetMetaDataAnnotation(&backupEntry.ObjectMeta, v1beta1constants.GardenerOperation, operation)
 		metav1.SetMetaDataAnnotation(&backupEntry.ObjectMeta, v1beta1constants.GardenerTimestamp, TimeNow().UTC().String())
+
 		if b.values.ShootPurpose != nil {
 			metav1.SetMetaDataAnnotation(&backupEntry.ObjectMeta, v1beta1constants.ShootPurpose, string(*b.values.ShootPurpose))
 		}
 
-		finalizers := sets.NewString(backupEntry.GetFinalizers()...)
-		finalizers.Insert(gardencorev1beta1.GardenerName)
-		backupEntry.SetFinalizers(finalizers.UnsortedList())
```

Comment:
Why was this removed?

Reply:
The backupentry controller adds its finalizer to the object [here](https://github.com/gardener/gardener/blob/b721f7f118f2d4cdb33ee0020f213a9fab6078eb/pkg/gardenlet/controller/backupentry/backup_entry_reconciler.go#L101), so there shouldn't be the need to add it here when creating the `BackupEntry` object. Also, this is semantically incorrect, as the finalizer should be added by the controller and not the applier, right?

Reply:
Ah, sure, yes, thanks for the explanation and for cleaning up!